infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -204,6 +203,27 @@ private void read(CorfuPayloadMsg<ReadRequest> msg, ChannelHandlerContext ctx, I
         }
     }
 
+    @ServerHandler(type = CorfuMsgType.MULTIPLE_READ_REQUEST, opTimer = metricsPrefix + "multiRead")
+    private void multiRead(CorfuPayloadMsg<MultipleReadRequest> msg, ChannelHandlerContext ctx, IServerRouter r,
+                           boolean isMetricsEnabled) {
+        log.trace("multiRead: {}", msg.getPayload().getAddresses());
+
+        ReadResponse rr = new ReadResponse();
+        try {
+            for (Long l : msg.getPayload().getAddresses()) {
+                ILogData e = dataCache.get(l);
+                if (e == null) {
+                    rr.put(l, LogData.EMPTY);
+                } else {
+                    rr.put(l, (LogData) e);
+                }
+            }
+            r.sendResponse(ctx, msg, CorfuMsgType.READ_RESPONSE.payloadMsg(rr));
+        } catch (DataCorruptionException e) {
```

Comment:
IOExceptions are wrapped by a runtime exception. If a runtime exception is thrown, the client would timeout and retry.